src/lib.rs

```diff
@@ -168,6 +168,37 @@ where
                 let x = <Point<T> as FromTokens<T>>::from_tokens_with_parens(tokens);
                 x.map(|y| y.as_item())
             }
+            w if w.eq_ignore_ascii_case("POINTZ") => {
+                let x = <Point<T> as FromTokens<T>>::from_tokens_with_parens(tokens)?;
+                if let Some(coord) = &x.0 {
+                    if coord.z.is_none() {
+                        return Err("POINTZ must have a z-coordinate.");
+                    }
+                }
+                Ok(x.as_item())
+            }
+            w if w.eq_ignore_ascii_case("POINTM") => {
+                let mut x = <Point<T> as FromTokens<T>>::from_tokens_with_parens(tokens)?;
+                if let Some(coord) = &mut x.0 {
+                    if coord.z.is_none() {
+                        return Err("POINTM must have a linear referencing system.");
```

Comment:
I couldn't find this in a spec, but "must have an M coordinate" (or "m-coordinate", but let's be consistent) seems more appropriate?

(same for `POINTZM`)

Reply:
From [wikipedia](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry)

> Coordinates for geometries may be 2D (x, y), [3D](https://en.wikipedia.org/wiki/Three-dimensional_space) (x, y, z), 4D (x, y, z, m) with an m value that is part of a [linear referencing system](https://en.wikipedia.org/wiki/Linear_referencing_system) or 2D with an m value (x, y, m).

Reply:
Yeah, but https://postgis.net/docs/ST_MakePointM.html calls them M coordinates and file:///home/grayshade/Downloads/06-103r4_Implementation_Specification_for_Geographic_Information_-_Simple_feature_access_-_Part_1_Common_Architecture_v1.2.1.pdf says:

> In addition, each coordinate carries an "m" ordinate value that is part of some linear reference system.

> The ordinates x, y and z are spatial, and the ordinate m is a measure.

So I think calling it an "M-coordinate" is a little clearer.

Reply:
fixed